src/lib/lore.sh

```diff
@@ -532,29 +558,36 @@ function process_representative_patches()
     # To avoid duplication, check if patch has been processed as representative
     message_id="${patch_dict['message_id']}"
     [[ -n "${processed_representative_patches["$message_id"]}" ]] && continue
+    in_reply_to_message_id="${patch_dict['in_reply_to']}"
 
     # Assume that patch number 0 is always the representative as the cover letter
     if [[ "${patch_dict['number_in_series']}" == 0 ]]; then
       is_representative_patch=1
     # Assume that, when there is no patch number 0, number 1 is the representative
+    # if it is a root of a thread
+    elif [[ "${patch_dict['number_in_series']}" == 1 && -z "$in_reply_to_message_id" ]]; then
+      is_representative_patch=1
+    # Assume that, if 'In-Reply-To' is not patch number 0 from the same version,
+    # number 1 is the representative
     elif [[ "${patch_dict['number_in_series']}" == 1 ]]; then
-      # Assume that patch number 1 without 'In-Reply-To' means no number 0
-      if [[ "${patch_dict['total_in_series']}" == 1 ||
-        -z "${patch_dict['in_reply_to']}" ]]; then
-        is_representative_patch=1
-      else
-        patch_metadata=()
-        in_reply_to_metadata=()
-        in_reply_to_message_id="${patch_dict['in_reply_to']}"
-        IFS=',' read -ra patch_metadata <<< "${individual_patches_metadata["$message_id"]}"
-        IFS=',' read -ra in_reply_to_metadata <<< "${individual_patches_metadata["$in_reply_to_message_id"]}"
+      patch_metadata=()
+      in_reply_to_metadata=()
+      IFS=',' read -ra patch_metadata <<< "${individual_patches_metadata["$message_id"]}"
 
-        # Assume that, if 'In-Reply-To' is not patch number 0 from the same
-        # version, number 1 is the representative
-        if [[ "${patch_metadata[0]}" != "${in_reply_to_metadata[0]}" || "${in_reply_to_metadata[1]}" != 0 ]]; then
-          is_representative_patch=1
+      if [[ -n "${individual_patches_metadata["$in_reply_to_message_id"]}" ]]; then
+        IFS=',' read -ra in_reply_to_metadata <<< "${individual_patches_metadata["$in_reply_to_message_id"]}"
+      else
+        in_reply_to_title=$(get_raw_lore_message "$message_id" | grep --perl-regexp '^Subject:' | sed 's/^Subject: //')
```

Comment:
Is it possible that the subject information should be two lines long with or without \n? If so, this grep/sed will work?

Reply:
Hey @rodrigosiqueira, I experimented a little more with some lists, and I am confident that this heuristic is valid. We really are only fetching the first line of `Subject:`, i.e., with multi-line subjects in the raw patch, the `grep`  command only gets its first line.

This is OK, because we assume that the patch tag will surely be in the first line (which I "confirmed" in my "experiments"), and, considering that we are already handling an edge case (we only fetch the raw patch, because we fell in the case of a patch number 1 that has an in reply to a message that wasn't processed yet), I think we are safe.

Anyway, give me your thoughts on this.

Reply:
I think we are fine!